src/ex2/input.js

```diff
@@ -0,0 +1,53 @@
+class Input {
+  constructor() {
+    this.inputButton = null;
+  }
+
+  checkInputType(input) {
+    let newPokemonsIdArr = [];
+    let newNormalTasks = [];
+    if (isTextContainOnlyNumbersAndSpaces(input)) {
+      newPokemonsIdArr.push(input);
+    } else if (input.includes(",")) {
+      let arr = input.split(",");
+      newPokemonsIdArr = arr.filter(isTextContainOnlyNumbersAndSpaces);
+
+      newNormalTasks = arr.filter((text) => {
+        return !isTextContainOnlyNumbersAndSpaces(text);
+      });
+    } else {
+      newNormalTasks.push(input);
+    }
+
+    return { newPokemonsIdArr, newNormalTasks };
+  }
+
+  getInputFromUser() {
+    const text = this.inputButton.value;
+    this.inputButton.value = ""; // nees to do it on the htmlEditor class
+    if (!text) {
+      return null;
+    }
+    return text;
+  }
+
+  handleAndValidateInput() {
+    let text = this.getInputFromUser();
+    if (!text) {
+      return null;
+    }
+
+    let { newPokemonsIdArr, newNormalTasks } = this.checkInputType(text);
+
+    newPokemonsIdArr = newPokemonsIdArr.map(clearSpacesFromNumber);
+
+    newNormalTasks = newNormalTasks.map(removeExtraSpaceFromTask);
```

Comment:
from removing extra spaces from the start & end of a string you can use `.trim()` function.
your function also handles removing extra spaces inside a task, but why? (maybe user intended to write it?)